content/articles/flutter_speech_recognition_using_alan_voice/index.md

```diff
@@ -0,0 +1,228 @@
+
+To achieve this we will first set up our flutter application, then we will link it to alan voice by integrating alan voice plugin into the application. Next we will test our speech recognition application. Finally we will create/edit/update our commands without updating our application.
+
+
+### Prerequisite
+For this tutorial to take effect, the user must have knowledge of how to create flutter applications or must have created a flutter application before.
+
+### Table of Contents
+- [Introduction](#introduction)
+- [Prerequisite](#prerequisite)
+- [Table of Contents](#table-of-contents)
+- [Goals](#goals)
+- [Setting up our application](#setting-up-our-application)
+- [Creating our Account](#creating-our-account)
+- [Running our application](#running-our-application)
+- [Conclusion](#conclusion)
+- [Further Reference](#further-reference)
+
+### Goals
+We will learn how to setup speech recognition using Alan voice plugin in our flutter application
+See how to set commands and call them in our application inorder to perform some operations.
+
+### Setting up our application
+Head over to your flutter application IDE and spin up a new flutter application and name it any suitable name of your choosing.
+Once your application is set and ready to run we need to integrate flutter alan voice package into our application, this can be obtained from [Flutter package alan_voice](https://pub.dev/packages/alan_voice). Get the latest version of the package and add it to your pubspec.yml file. Ensure the minimumSdkVersion for your application is 21 to check this you can update this in your android/app/build.gradle file.
+
+![set-up-application](/flutter_speech_recognition_using_alan_voice/application.png)
+
+
+
+### Creating our Account 
+We need to set up our account to be able to configure our application commands without updating codes in our application. To achieve this we need to link our application to our alan voice account we just created by using an Alan SDK key provided to us upon account creation. Head over to [studio alan](https://studio.alan.app/) and create an account with your details. Click on create voice assistance and set up a new project. Once done you should see a basic “Hello world” - “Hi there” reply described in the center window panel. 
+
+
+![create_account](/flutter_speech_recognition_using_alan_voice/account.png)
+
+
+Inorder to get our Alan SDK key click on the integration button on the top navigation on the window. Select the platform you want to integrate to ours is android so we select android. Note that for and ios application we need to obtain the configuration for ios by selecting the IOS option
+
+
+### Running our application
+We import our plugin file into our application and we have in our main.dart file the following code
+
+```dart
```

Comment:
Please properly explain all your code below each block and add an in-line comment if necessary.